spec/views/users/show.html.erb_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show Page', type: :feature do
+  before do
+    @user = create(:user, :with_posts)
+    @most_recent_posts = @user.posts.order(created_at: :desc).limit(3)
+  end
+
+  context 'Displaying User Information' do
+    before { visit user_path(@user) }
+
+    it 'displays the user\'s profile picture' do
+      expect(page).to have_css('.photo img')
+    end
+
+    it 'displays the user\'s username' do
+      expect(page).to have_content(@user.name)
+    end
+
+    it 'displays the number of posts the user has written' do
+      expect(page).to have_content("Number of Posts: #{@user.posts.count}")
+    end
+
+    it 'displays the user\'s bio' do
+      expect(page).to have_content(@user.bio)
+    end
+
+    it 'displays the user\'s first 3 posts' do
+      @most_recent_posts.each do |post|
+        expect(page).to have_content("Post ##{post.id}")
+        expect(page).to have_content(post.text)
+        expect(page).to have_content("comments: #{post.comments_counter}, Likes: #{post.likes_counter}")
+      end
+    end
+  end
+
+  context 'Interactions' do
+    before { visit user_path(@user) }
+
+    it 'redirects to the user\'s posts index page when clicking "Show More" button' do
+      click_link 'Show More'
+      expect(page).to have_current_path(user_posts_path(@user))
+    end
+  end
```

Comment:
@joel-kalema ,
thank you for the review but this requested added test were implemented above.
this test to be specific:
- I can see a button that lets me view all of a user's posts.

Reply:
@rugwizangoga 
Keep doing good job 🥇